src/dashboard/src/main/migrations/0031_dashboardsetting_scope.py

```diff
@@ -0,0 +1,238 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import ast
+
+from django.db import migrations, models
+
+from main.models import DashboardSetting, Job
+
+
+def data_migration_atom_to_dict(apps, schema_editor):
+    """
+    Convert old settings into a new DashboardSetting dict, e.g. in the
+    scenario where the user had the following settings in the previous form:
+
+        +--------------------------+-----------------------+
+        | name                     | value                 |
+        +--------------------------+-----------------------+
+        | dip_upload_atom_url      | http://localhost/atom |
+        | dip_upload_atom_email    | demo@example.com      |
+        | dip_upload_atom_password | demo                  |
+        | dip_upload_atom_version  | 2                     |
+        +--------------------------+-----------------------+
+
+    We're going to convert them into:
+
+        +---------------+-----------------------+-------------------+
+        | name          | value                 | scope             |
+        +---------------+-----------------------+-------------------+
+        | url           | http://localhost/atom | upload-qubit_v0.0 |
+        | version       | 2                     | upload-qubit_v0.0 |
+        | password      | demo                  | upload-qubit_v0.0 |
+        | email         | demo@example.com      | upload-qubit_v0.0 |
+        | debug         |                       | upload-qubit_v0.0 |
+        | rsync_target  |                       | upload-qubit_v0.0 |
+        | rsync_command |                       | upload-qubit_v0.0 |
+        | key           |                       | upload-qubit_v0.0 |
+        +---------------+-----------------------+-------------------+
+                                        * Blank means empty string.
+
+    """
+
+    old_prefix = 'dip_upload_atom_'
+    fields = ('url', 'email', 'password', 'rsync_target', 'rsync_command', 'version', 'debug', 'key')
+
+    qs = apps.get_model('main', 'DashboardSetting').objects.filter(name__in=['{}{}'.format(old_prefix, field) for field in fields])
+    old_dict = dict(qs.values_list('name', 'value'))
+    new_dict = {field: old_dict.get('{}{}'.format(old_prefix, field)) for field in fields}
+
+    # Set new dict and delete previous tuples. Reminder: migrations run inside
+    # a transaction.
+    DashboardSetting.objects.set_dict('upload-qubit_v0.0', new_dict)
```

Comment:
I'm glad you've asked this question because I hadn't thought about it!

I've found two relevant sections in the docs:
- [Historical models](https://docs.djangoproject.com/en/1.8/topics/migrations/#historical-models)
- [Model managers](https://docs.djangoproject.com/en/1.8/topics/migrations/#model-managers)

If I understood properly I can add the attribute `use_in_migrations` (`True`) to `DashboardSetting` so the manager becomes part of the migration. However, the manage has to stay around as long as the migration references to the manager. The method `set_dict` should be compatible with all past migrations using it.

More here: https://code.djangoproject.com/ticket/23822.

I could take the two following actions:
- Use `use_in_migrations` (will probably require to regenerate the migration).
- Add a warning to the docs of `set_dict` mention this issue.

Thoughts?